superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx

```diff
@@ -410,18 +420,22 @@ function DashboardList(props: DashboardListProps) {
         { label: t('Unpublished'), value: false },
       ],
     },
-    {
-      Header: t('Favorite'),
-      id: 'id',
-      urlDisplay: 'favorite',
-      input: 'select',
-      operator: FilterOperators.dashboardIsFav,
-      unfilteredLabel: t('Any'),
-      selects: [
-        { label: t('Yes'), value: true },
-        { label: t('No'), value: false },
-      ],
-    },
+    ...(props.user.userId
+      ? [
+          {
+            Header: t('Favorite'),
+            id: 'id',
+            urlDisplay: 'favorite',
+            input: 'select',
+            operator: FilterOperators.dashboardIsFav,
+            unfilteredLabel: t('Any'),
+            selects: [
+              { label: t('Yes'), value: true },
+              { label: t('No'), value: false },
+            ],
+          } as Filter,
```

Comment:
Is there any way we can cleanly avoid using `as Filter` here? Ideally the type system would know that this object is compatible without a type assertion.